frontend/src/service/Url.service.ts

```diff
@@ -151,8 +151,8 @@ export class UrlService {
         .catch(({ graphQLErrors, networkError, message }) => {
           console.log(graphQLErrors);
           const errCodes = graphQLErrors.map(
-            (graphQLError: GraphQlError) => graphQLError.extensions.code
-          );
+            (graphQLError: GraphQlError) => (graphQLError.extensions) ? graphQLError.extensions.code : null
```

Comment:
```suggestion
            graphQLError: GraphQlError => (graphQLError.extensions) ? graphQLError.extensions.code : null
```

Please add the following above line 153 to handle network error case.

```typescript
if (!graphQLErrors) {
  return reject([networkError]);
}
```

Can we add `Unknown` at [here](https://github.com/byliuyang/short/blob/1aa5c474c273362fb0c8a1a78ad324f1c1b3bdef/frontend/src/service/Error.service.ts#L6-L7) and use enum instead of null at line 154? Even though `extensions` is empty, it's still an graphQL error.